src/control/server/ctl_network.go

```diff
@@ -0,0 +1,49 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package server
+
+import (
+	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/daos-stack/daos/src/control/server/ioserver"
+)
+
+// NetworkScanService encapsulates the network part of the control service
+type NetworkScanService struct {
+	nsslog  logging.Logger
```

Comment:
The problem I had was a namespace collision with the log in the other service.  I gave it a separate name to avoid that.  What do you recommend to eliminate the namespace collision?

Reply:
I would recommend just adding these methods directly to the ControlService struct. It's fine to keep them in the files they're in, for organization. The only reason that StorageControlService exists is so that the daos_server subcommands can import and use its methods without the rest of the ControlService baggage. As your new network code is only called via RPC there's not a good reason to split it out into an embeddable struct as far as I can see.

Reply:
done.